src/mito2/src/sst/file_purger.rs

```diff
@@ -143,6 +160,29 @@ impl FilePurger for LocalFilePurger {
     }
 }
 
+impl FilePurger for LocalFilePurger {
+    fn send_request(&self, request: PurgeRequest) {
+        if self.is_local_fs {
+            if request.deleted {
+                self.delete_file(request.file_meta);
+            }
+        } else {
+            // if not on local file system, instead inform the global file purger to remove the file reference.
+            // notice that no matter whether the file is deleted or not, we need to remove the reference
+            // because the file is no longer in use nonetheless.
+            self.file_ref_manager.remove_file(&request.file_meta);
+        }
+    }
+
+    fn add_new_file(&self, file_meta: &FileMeta) {
+        if self.is_local_fs {
+            // If the access layer is local file system, we don't need to track the file reference.
+            return;
+        }
+        self.file_ref_manager.add_file(file_meta);
+    }
+}
```

Comment:
Should we implement two separate `FilePurgers` instead of mixing them together? For example, `LocalFsFilePurger` and `ObjectStoreFilePurger`.